lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -870,6 +870,21 @@ tools::ParsePICArgs(const ToolChain &ToolChain, const ArgList &Args) {
     PIP = true;
   }
 
+  // Pagerando and '-ffunction-sections'/'-fdata-sections' are incompatible
+  bool UseSeparateSections = isUseSeparateSections(Triple);
+  bool FunctionSections = Args.hasFlag(options::OPT_ffunction_sections,
+                                       options::OPT_fno_function_sections,
+                                       UseSeparateSections);
+  bool DataSection = Args.hasFlag(options::OPT_fdata_sections,
+                                  options::OPT_fno_data_sections,
+                                  UseSeparateSections);
+  if (PIP && (FunctionSections || DataSection))
```

Comment:
We should document somewhere _why_ pagerando is not compatible with function/data sections. It might actually be nice to still emit individual function sections for non-pagerando functions, but that's really not an important case imo. Data sections are more important, and work fine for Arm32. `-fdata-sections` shouldn't affect correctness when combined with pagerando, so maybe we should consider allowing it?